src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -1612,12 +1612,12 @@ int Deoptimization::reassign_fields_by_klass(InstanceKlass* klass, frame* fr, Re
       intptr_t val;
       ScopeValue* scope_field = sv->field_at(svIndex);
       StackValue* value = StackValue::create_stack_value(fr, reg_map, scope_field);
-      offset += j * type2aelembytes(type);
```

Comment:
Nice catch!!, It went in along with merge patch. Will run this through jtreg tests to ensure status quo or better validation status.

Best Regards,
Jatin

Reply:
Please also relax the condition on line # 1605 to just check for Location::vector

Reply:
Yes, relaxing the condition is fine. And we'd better to add the assertion that `secondary_fields_count > 1` ?

Reply:
Or, does the vector supported for 1 lanecount (e.g. 64-bit shape for `long/double` type) on x86 systems?

Reply:
> Or, is the vector supported for 1 lanecount (e.g. 64-bit shape for `long/double` type) on x86 systems?

Not yes, most of the vector intrinsic also fail for single lane vector operations.

Reply:
Yeah, I also met the bad ad file crashes due to single lane for double type on x86 system. And I tried to add the assertion that `secondary_fields_count > 1`, but it crashes when running `Double64VectorTests` on x86. Anyway, I have removed this condition and assertion in the deoptimization.